src/blockchain.rs

```diff
@@ -1,116 +1,100 @@
     }
 
-    /// Returns the latest `Block` as part of the longest chain
-    pub fn get_latest_block_index(&self) -> Option<&BlockIndex> {
-        self.index.blocks.last()
-    }
-
     /// Append `Block` to the index of `Blockchain`
-    ///
-    /// * `block` - `Block` appended to index
     pub fn add_block(&mut self, block: Block) {
-        for tx in block.transactions().iter() {
-            self.shashmap.insert_new_transaction(tx);
-        }
-
-        let block_index: BlockIndex = (block.header().clone(), block.hash());
-        println!("{:?}", block_index.clone());
-        self.index.blocks.push(block_index);
+        self.blocks.push(block);
     }
+
+    // /// Return latest `Block` in blockchain
+    // pub fn get_latest_block(&mut self) -> &Block {
+    // 
+    // }
 }
 
+
 #[cfg(test)]
 mod tests {
 
-    use super::*;
-    use crate::block::Block;
-    use crate::keypair::Keypair;
-    use crate::slip::{Slip, SlipBroadcastType};
-    use crate::transaction::{Transaction, TransactionBroadcastType};
-
-    #[test]
-    fn blockchain_test() {
-        let blockchain = Blockchain::new();
-        assert_eq!(blockchain.index.blocks, vec![]);
-    }
-    #[test]
-    fn blockchain_get_latest_block_index_none_test() {
-        let blockchain = Blockchain::new();
-        match blockchain.get_latest_block_index() {
-            None => assert!(true),
-            _ => assert!(false),
-        }
-    }
-    #[test]
-    fn blockchain_get_latest_block_index_some_test() {
-        let mut blockchain = Blockchain::new();
-        let block = Block::new(Keypair::new().public_key().clone(), [0; 32]);
-
-        blockchain.add_block(block.clone());
-
-        match blockchain.get_latest_block_index() {
-            Some((prev_block_header, _)) => {
-                assert_eq!(&prev_block_header.clone(), block.header());
-                assert!(true);
-            }
-            None => assert!(false),
-        }
-    }
-    #[test]
-    fn blockchain_add_block_test() {
-        let keypair = Keypair::new();
-        let mut blockchain = Blockchain::new();
-        let mut block = Block::new(keypair.public_key().clone(), [0; 32]);
-        let mut transaction = Transaction::new(TransactionBroadcastType::Normal);
-        let slip = Slip::new(
-            *keypair.public_key(),
-            SlipBroadcastType::Normal,
-            2_0000_0000,
-        );
-        transaction.add_output(slip);
-        block.add_transaction(transaction);
+    // use super::*;
+    // use crate::block::Block;
+    // use crate::keypair::Keypair;
+    // use crate::slip::{OutputSlip, SlipID};
+    // use crate::transaction::{Transaction, TransactionType};
+    // use secp256k1::Signature;
 
-        blockchain.add_block(block.clone());
-        let (block_header, _) = blockchain.index.blocks[0].clone();
-
-        assert_eq!(block_header, *block.clone().header());
-        assert_eq!(blockchain.shashmap.slip_block_id(&slip), Some(&-1));
-    }
+    // #[test]
```

Comment:
Instead of commenting out tests you could use `ignore`:  
https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute

Reply:
Okay, but they might not even build at the moment... I'll open up a new brnach and try to get things cleaned up today.